packages/backend/src/analytics/LightdashAnalytics.ts

```diff
@@ -665,6 +667,66 @@ type DashboardView = BaseTrack & {
     };
 };
 
+type ViewSqlChart = BaseTrack & {
+    event: 'sql_chart.view';
+    userId: string;
+    properties: {
+        chartId: string;
+        projectId: string;
+        organizationId: string;
+    };
+};
+
+type CreateSqlChartEvent = BaseTrack & {
+    event: 'sql_chart.created';
+    userId: string;
+    properties: {
+        chartId: string;
+        projectId: string;
+        organizationId: string;
+    };
+};
+
+type UpdateSqlChartEvent = BaseTrack & {
+    event: 'sql_chart.updated';
+    userId: string;
+    properties: {
+        chartId: string;
+        projectId: string;
+        organizationId: string;
+    };
+};
+
+type DeleteSqlChartEvent = BaseTrack & {
+    event: 'sql_chart.deleted';
+    userId: string;
+    properties: {
+        chartId: string;
+        projectId: string;
+        organizationId: string;
+    };
+};
+
+export type CreateSqlChartVersionEvent = BaseTrack & {
+    event: 'sql_chart_version.created';
+    userId: string;
+    properties: {
+        chartId: string;
+        versionId: string;
+        projectId: string;
+        organizationId: string;
+        chartKind: ChartKind;
+        barChart?: {
+            groupByCount: number;
+            yAxisCount: number;
+            aggregationTypes: string[];
+        };
+        pieChart?: {
+            groupByCount: number;
+        };
```

Comment:
Do we want the table viz here too or are you just not including more data about it because there's not a lot?

Reply:
Table config doesn't have groupBy and I wasn't sure what was worth tracking for now. I asked the team about it and we can update later. https://lightdash.slack.com/archives/C02FSS5HL2E/p1722499755037229?thread_ts=1722432671.482009&cid=C02FSS5HL2E